src/permutat.cc

```diff
@@ -1346,27 +1346,21 @@ static inline Obj CYCLE_STRUCT_PERM(Obj perm)
     ende = cnt;
 
     /* create the list */
-    list = NEW_PLIST(T_PLIST, max - 1);
-    SET_LEN_PLIST(list, max - 1);
+    list = NEW_PLIST(T_PLIST, max);
+    SET_LEN_PLIST(list, max);
```

Comment:
Until here, it seems you just changed `len` and `max` to be 1 smaller than they were before, yes? (This is of course fine, I am just trying to verify I understand your changes correctly)